website/integrations: add engomo (#10538)
goauthentik/authentik: website/integrations/services/engomo/index.md

---

> title: Engomo Composer + App Login

Author: I wanted to clear this out, that you are able to achieve the admin webgui login (composer) with different settings as you need for the App Login. Different redirect URI and an additional scope is needed.
This was/is making some trouble with an other Enterprise IDP I have to work with. So with authentik both is possible easy peasy :) At least it isn't neccessary to point this out at the end for the rest of the world.... I guess most of them (whoever they are) will just look for engomo and expect that everything works with this guide right :)

---

<span class="badge badge--secondary">Support level: Community</span>

## What is Engomo

> > Engomo is an enterprise application to create apps for smartphones/tablets based on Android, iOS or iPadOS.

Reviewer: ```suggestion
> engomo is an low-code app development platform to create enterprise apps for smartphones and tablets based on Android, iOS, or iPadOS.
```

> -- https://engomo.com/
>

> > This guide explains how to setup a Engomo to use authentik as OAUTH provider for the application login on the smartphone/tablet and login to the admin WebGUI (composer).

Reviewer: ```suggestion
> This guide explains how to set up engomo to use authentik as the OAUTH provider for the application login on the smartphone/tablet and login to the admin WebGUI (composer).
```

## Preparation

The following placeholders will be used:

-   `engomo.company` is the FQDN of the Engomo install.
-   `authentik.company` is the FQDN of the authentik install.
-   `engomo.mapping` is the name of the Scope Mapping.
-   `ak.cert` = The authentik self-signed certificate you use for the service provider.

> ## Custom Property Mapping

Reviewer: ```suggestion
## authentik configuration
```

> Create a new Scope Mapping under the Customization settings (Property Mapping).

Reviewer: ```suggestion
In authentik, create a new scope mapping. To do so, log in to the Admin interface, and navigate to **Customization-->Property Mapping** and then click **Create**.
```

-   `engomo.mapping` is the value for the Name.
-   `profile` is the value for the Scope name.
-   `return {"preferred_username": request.user.email,}` is the value for the Expression.

> Create an application and OAuth2/OpenID Provider in authentik. Create a OAuth2/OpenID Provider with the following parameters:

Reviewer: ```suggestion
Create an application and OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider::
```

Provider:

-   Name: `SP-engomo`
-   Client type: `Public`    
-   Redirect URIs/Origins (RegEx) #1: `https://engomo.company/auth` (for composer)
-   Redirect URIs/Origins (RegEx) #2: `com.engomo.engomo://callback/` (for app)
-   Signing Key: `ak.cert`
-   Scopes: `authentik default OAuth Mapping: OpenID 'email'`
-   Scopes: `authentik default OAuth Mapping: OpenID 'offline_access'`
-   Scopes: `authentik default OAuth Mapping: OpenID 'openid'`
-   Scopes: `engomo.mapping`
-   Note the Client ID and Client Secret.

> [!IMPORTANT]
> Redirect URIs => write the values line by line.

> Lave the rest as default values. You can of course adjust durations.

Reviewer: ```suggestion
Leave the rest as default values. You can of course adjust durations.
```

Application:

-   Name: `Engomo`
-   Slug: `engomo`
-   Launch URL: `https://engomo.company/`

> ## Engomo Configuration

Reviewer: ```suggestion
## engomo configuration
```

> Navigate to `https://engomo.company/composer` and login with your admin credentials.

Reviewer: ```suggestion
Navigate to `https://engomo.company/composer` and log in with your admin credentials.
```

Reviewer: When used as a verb, it's two words, log in.

-   Select `Server`.
-   Select `Authentication`.
-   Add a new authentcation method by clicking on the plus icon on the right.
-   Name: `authentik`
-   Type: `OpenID Connect`

> -   Click on Create.

Reviewer: ```suggestion
-   Click **Create**.
```

-   Set the `Issuer` to the authentik FQDN `https://authentik.company/application/o/engomo`.

> -   Set the `Client ID` to the Client ID from the SP-engomo provider inside authentik.

Reviewer: ```suggestion
-   Set the `Client ID` to the Client ID from the SP-engomo provider that you created in authentik.
```

> -   Set the `Client Secret` to the Client Secret from the SP-engomo provider inside authentik.

Reviewer: ```suggestion
-   Set the `Client Secret` to the Client Secret from the SP-engomo provider that you created in authentik.
```

Leave the rest as default.

> ## Engomo User Creation

Reviewer: ```suggestion
## engomo user creation
```

> Engomo doesn't create users automatically on signing in. So you have to do it manually right now.

Reviewer: ```suggestion
engomo doesn't create users automatically on signing in. So you have to do it manually right now.
```

> Navigate to `https://engomo.company/composer` and login with your admin credentials.

Reviewer: ```suggestion
Navigate to `https://engomo.company/composer` and log in with your admin credentials.
```

- Select `Users & Devices`.
- Click the plus button next in the Users section.
- Select `authentik` as the Authenticator in the dropdown.

> - Create your user by typing in the email as the Username used in authentik.

Reviewer: Does this mean "Create your user by typing in the email address that you defined as the Username in authentik." ?

I'm not sure I follow it, the way it is written.

Also, if there are there any validation steps that one can take, or next steps, it would be good to have a wrap up sentence at end. Maybe something like "Now your engomo apps are configured to use authentik for login authentication." or some such.

Author: Thais Right. The admin ahpuld cremte Users with their corresponding email from authentik to be able to use it and leg the Users Log in.

Will do those steps later. Learned more about english 😎 First I have to figure out on how to run the scripts inside WebGUI of github.

Useally I only fork a repo and then edit it in the browser :) there are 1 or 2 questions I have to ask the engomo Support before I can answer here or accept a Commit or change.

Capital and small letters are made with my Smartphone in this answer, sry for that 😩

Reviewer: Yeah, you can't run the linters from the GitHub web UI ... Here are other options to get the `prettier` lint run:
1. I would encourage you to fork and clone our repo to your local environment, and then use a terminal. It's a great way to practice the basics of Git and become comfortable with command lines. I personally think CLIs are way more fun than UIs, but don't let my bias sway you, haha. Or, if you fork and clone to a local repo, you can use one of the many third-party git management tools with a UI.
2. I can do it, and push the fix commit to your PR branch

We appreciate this contribution! Let me know how you'd like to proceed.

Author: I wrote you an email to get this done in any way, want to learn this stuff :)
Btw: Thank you very much for reviewing my PR! You did awesome job on my FortiGate things months ago! I appreciate it.

> -- https://engomo.com/
>
> This guide explains how to set up engomo to use authentik as the OAUTH provider for the application login on the smartphone/tablet and login to the admin WebGUI (composer).

## Preparation

The following placeholders will be used:

-   `engomo.company` is the FQDN of the Engomo install.
-   `authentik.company` is the FQDN of the authentik install.
-   `engomo.mapping` is the name of the Scope Mapping.
-   `ak.cert` = The authentik self-signed certificate you use for the service provider.

## authentik configuration

In authentik, create a new scope mapping. To do so, log in to the Admin interface, and navigate to **Customization-->Property Mapping** and then click **Create**.

-   `engomo.mapping` is the value for the Name.
-   `profile` is the value for the Scope name.
-   `return {"preferred_username": request.user.email,}` is the value for the Expression.

Create an application and OAuth2/OpenID provider in authentik. Use the following parameters for the OAuth2/OpenID provider::

Provider:

-   Name: `SP-engomo`
-   Client type: `Public`    
-   Redirect URIs/Origins (RegEx): `https://engomo.company/auth` and `com.engomo.engomo://callback/`

> -   Signing Key: `ak.cert`

Author: Don't know how/what to do with this. Do you mean the name? I can use anything more "standardized" if you want. All my SP's got the `SP-` in front of the apps name.

-   Scopes: `authentik default OAuth Mapping: OpenID 'email'` and `authentik default OAuth Mapping: OpenID 'offline_access'` and `authentik default OAuth Mapping: OpenID 'openid'` and `engomo.mapping`
-   Note the Client ID and Client Secret.

> > [!IMPORTANT]

Author: What do you mean? The Scopes?

We struggled in my company with a huge firewalling company that provides IDP as well (that we have at work). There we figured out that their IDP isn't using any standard scopes. You always have to add them one by one and only all 4 of them did work.

I also got in contact with the dev/support guys of engomo. They told me exactly what their application needs.
What I think about right now is that we don't need the certificate... at least I guess so. Will check and report back.

> > [!IMPORTANT]

Author: > When looking at review comments look at the line before the comment starts. in this case it would be the IMPORTANT and that block.
> 
> I was saying that i'm not sure the block is needed as it can be guessed and the notice is not present in other integration pages.

so I comma seperated now and remove the "note line" ok?

> Redirect URIs => write the values line by line.

Leave the rest as default values. You can of course adjust durations.

> Application:

Author: First thank you for your help :)
What do you mean with this?

something like this?
```
- Rest of settings: Leave the rest as default values.....
```

> Application:

Author: so you mean I should do the `Application:` as this?

### Application

If so, no problem will do it, but it isn't the way I was doing it on my old PR that was merged in the past.

## engomo configuration

Navigate to `https://engomo.company/composer` and log in with your admin credentials.

-   Select `Server`.
-   Select `Authentication`.
-   Add a new authentication method by clicking on the plus icon on the right.
-   Name: `authentik`
-   Type: `OpenID Connect`
-   Click **Create**.
-   Set the `Issuer` to the authentik FQDN `https://authentik.company/application/o/engomo`.
-   Set the `Client ID` to the Client ID from the SP-engomo provider that you created in authentik.
-   Set the `Client Secret` to the Client Secret from the SP-engomo provider that you created in authentik.

Leave the rest as default.

## engomo user creation

engomo doesn't create users automatically on signing in. So you have to do it manually right now.
Navigate to `https://engomo.company/composer` and log in with your admin credentials.

- Select `Users & Devices`.
- Click the plus button next in the Users section.
- Select `authentik` as the Authenticator in the dropdown.
- Create your user by typing in the email as the Username used in authentik.

At this point you are done.

## testing the login

- Open a browser of your choice and open the URL `https://engomo.company`.

> - Enter the created user's email adress and hit the small arrow icon to log in.

Reviewer: ```suggestion
- Enter the created user's email address and hit the small arrow icon to log in.
```

- You should be redirected to authentik with all its login flows you have created by yourself.

> - If you are redirected back to the `https://engomo.company/composer` URL you did everything correct.

Reviewer: Hmmmm... so is it that the user is first redirected back to authentik, and then on to https://engomo.company/composer?

If Yes, maybe clarify by saying something like:

"You should be redirected to authentik (via the with login flows you created) and then authentik should redirect you back to the `https://engomo.company/composer` URL."

Reviewer: Also, thanks for adding these verification steps!!

Author: done

> [!IMPORTANT]

> > You will only have access to the app or composer page if you granted the permission to the newly created user of course.

Reviewer: Talking about both "You" and the "new user" is a little confusing... is that the same person? Do you mean "The user will only have access to the app or composer page if you granted the permission to them, of course."

And... do you mean granting permission to them in authentik or in that app or... where exactly?

Thanks for your patience with clarifying these kinda details. Communication is hard. ;-)

Author: everything is fine, at least I have to thank you and this community here :) This is an application that most here won't have access to. That makes it even harder :) Fortunately OIDC/OAuth is a standard and that makes it "usually the same"... 

I have one question, I guess it is not allowed to use screenshots in the docs right? Was thinking about of posting them but this will be the first integration that has them.
